### DISCUSSION TITLE
Is it possible to apply AllenCahn kernel to vector variables?

### DISCUSSION THREAD

**Discussion by RWTHLHK** [idaholab/moose#27131] [Q&A Modules: Phase field]
I am developing a new phase field model. It requires the evolution of an ordered vector parameter with 12 components. Each component of this vector obeys the Allen-Cahn equation. Using 12 coupled variables in the input file can do this, but it is super inconvenient. Is there any other way to do this?

**Comment by GiudGiud** [comment]
Hello

If the equations are very similar, then you can use array variables for this. One variable with 12 components.
@laagesen do we have those for AC?

Guillaume

**Reply by laagesen** [reply]
Hi @RWTHLHK - We normally use Actions to automate adding multiple variables and kernels for multiple Allen-Cahn equations. Have a look for example at the page https://mooseframework.inl.gov/modules/phase_field/Grain_Growth_Model.html and in particular the section `Simplified MOOSE Syntax` and pages linked there. You may be able to use what is already there or modify those actions for your specific model.
